### PR TITLE
Add Support for Root Admins to Submit Tax Forms

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4805

⚠️ Require: https://github.com/opencollective/opencollective-api/pull/8701

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
This is going to be really useful!

It could have been nice to make the year customizable; one can update a tax form in 2023 for an expense created in 2022 (not a blocker).

File: components/expenses/TaxFormLinkModal.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+              try {
+                setError(null);
+                const result = await submit({
+                  variables: {
+                    account: { slug: account.slug },
+                    taxFormLink,
+                    year,
+                  },
+                });
+                if (get(result, 'data.setTaxForm.success')) {
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/expenses/TaxFormLinkModal.js:123]: I've added an error toast for this. 👍🏽

**Comment by SudharakaP** [comment]
> This is going to be really useful!
> 
> It could have been nice to make the year customizable; one can update a tax form in 2023 for an expense created in 2022 (not a blocker).

@Betree : I've added an year input as well. 👍🏽 

image

File: components/expenses/TaxFormLinkModal.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+              linkToTaxFormsFolder: (
+                <StyledLink
+                  href="https://drive.google.com/drive/folders/1ga_-6tBTqADvngRr9nbxRKf7yoSne1n1?usp=share_link"
+                  openInNewTab
+                >
+                  Google Drive Folder
+                </StyledLink>
+              ),
+            }}
+          />
```
Comment (Betree) [inline components/expenses/TaxFormLinkModal.js:71]: Great that you added this message; admins often don't know where to upload docs. Also Fyi, there's this unresolved situation: https://opencollective.slack.com/archives/GST5GVCQ7/p1674030322385029?thread_ts=1673992250.622439&cid=GST5GVCQ7.

Small change requested on the translation itself as it's incorrect, see https://docs.opencollective.com/help/contributing/development/translations#translate-links-inline.

```suggestion
          <FormattedMessage
            defaultMessage="Please upload the tax form to the <Link>Google Drive Folder</Link>. Copy the link to the file and paste it in the field below."
            values={{
              linkToTaxFormsFolder: getI18nLink({ href: 'https://drive.google.com/drive/folders/1ga_-6tBTqADvngRr9nbxRKf7yoSne1n1?usp=share_link', openInNewTab: true }}
          />
```

File: components/expenses/TaxFormLinkModal.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+const arrayOfYears = () => {
+  const max = new Date().getFullYear();
+  const min = 2018;
+  const years = [];
+
+  for (let i = max; i >= min; i--) {
+    years.push(i);
+  }
+  return years;
```
Comment (Betree) [inline components/expenses/TaxFormLinkModal.js:39]: :ice_cream: lodash to the rescue

```suggestion
  return range(2018, new Date().getFullYear());
```

**Review by Betree — COMMENTED** [review]
Looks good to me; only a small requested change on the i18n string. Feel free to merge once it's fixed :rocket:

File: components/expenses/TaxFormLinkModal.js (hunk 3 above)
Comment (SudharakaP) [reply components/expenses/TaxFormLinkModal.js:39]: Thanks for the simplification. 👍🏽

File: components/expenses/TaxFormLinkModal.js (hunk 2 above)
Comment (SudharakaP) [reply components/expenses/TaxFormLinkModal.js:71]: @Betree : I've done the translation change. 


> Great that you added this message; admins often don't know where to upload docs. Also Fyi, there's this unresolved situation: https://opencollective.slack.com/archives/GST5GVCQ7/p1674030322385029?thread_ts=1673992250.622439&cid=GST5GVCQ7.

The link here seems broken. Could you please tell me what this is about?

**Pull request merged**